etc/systemd/system/zfs-share.service.in

```diff
@@ -14,6 +14,7 @@ ConditionPathIsDirectory=/sys/module/zfs
 Type=oneshot
 RemainAfterExit=yes
 EnvironmentFile=-@initconfdir@/zfs
+ExecStartPre=-flock /etc/exports.d/zfs.exports.lock truncate --size 0 /etc/exports.d/zfs.exports
```

Comment:
This seems fine for linux but we should verify that FreeBSD does not suffer from the same issue and that we don't need a more complete fix.

Reply:
For FreeBSD:
```
zfs_start_main()
{
        zfs mount -va
        zfs share -a
        if [ ! -r /etc/zfs/exports ]; then
                touch /etc/zfs/exports
        fi
}
```
would become:
```
zfs_start_main()
{
        zfs mount -va
        truncate -s 0 /etc/zfs/exports
        zfs share -a
}
```

Reply:
refactored to make it work for any OS